modules/BuildUtilities/src/server/PrepareConfigFiles.coffee

```diff
@@ -186,23 +185,40 @@ getProperties = (configDir) =>
 		allConf = _.extend allConf, propertiesParser.read(configFile)
 		console.info "read conf file: #{configFile}"
 
-
 	configString = ""
+	lowerCaseConfNames = []
+	confNames = []
 	for attr, value of allConf
+		lowerCaseConfNames.push(attr.toLowerCase())
+		confNames.push(attr)
 		if value != null
 			configString += attr+"="+value+"\n"
 		else
 			configString += attr+"=\n"
 
-	envFiles = glob.sync("#{acasHome}/conf/*.env")
-	envFiles = envFiles.reverse()
-	envFiles.push "#{acasHome}/conf/.env"
-	console.info "started with process.env and now reading env files in this order (former envs take presidence over latter envs): #{envFiles}"
-	for envFile in envFiles
-		console.info "reading env file: #{envFile}"
-		fileEnv = dotenv.config({ path: envFile} )
-		dotenvexpand(fileEnv)
-		console.info "read env file: #{envFile}"
+	# Read environment variables prefixed with ACAS_ and substitute them in the config file
+	envinmentSubstitutions = {}
+	for key, value of process.env
+		if key.startsWith('ACAS_')
+			# Remove ACAS prefix and lower case
+			newKey = key.substring(5).toLowerCase()
+			# To deal with configs with an underscore, a user can set a double _ inside an environment variable and ACAS will replace it with a single _
+			# Replace double underscores with a unique separator
+			uniqueSeperator = '----------------------------ACAS-UNIQUE-SEPERATOR----------------------------'
```

Comment:
nit: typo in `seperator` should be `separator`

Reply:
Fixed